expression/builtin_encryption.go

```diff
@@ -407,13 +407,23 @@ func (b *builtinDecodeSig) evalString(row chunk.Row) (string, bool, error) {
 	if isNull || err != nil {
 		return "", true, err
 	}
+	dataTp := b.args[0].GetType()
+	dataBuf, err := charset.NewEncoding(dataTp.Charset).EncodeString(dataStr)
+	if err != nil {
+		return "", true, err
+	}
 
 	passwordStr, isNull, err := b.args[1].EvalString(b.ctx, row)
 	if isNull || err != nil {
 		return "", true, err
 	}
+	passwordTp := b.args[0].GetType()
```

Comment:
should be passwordTp := b.args[1].GetType()?

Reply:
You are right. Fixed.